diff -ruN nDPI/src/lib/protocols/ssl.c nDPI.new/src/lib/protocols/ssl.c
--- nDPI/src/lib/protocols/ssl.c	2016-11-16 21:40:09.335240761 +0300
+++ nDPI.new/src/lib/protocols/ssl.c	2016-11-16 21:39:50.960651510 +0300
@@ -114,6 +114,7 @@
     
     if((buffer[i] != '.')
        && (buffer[i] != '-')
+       && (buffer[i] != '_')
        && (buffer[i] != '*')
        && (!ndpi_isalpha(buffer[i]))
        && (!ndpi_isdigit(buffer[i]))) {
