--- nDPI.new/src/lib/protocols/ssl.c	2016-11-16 21:39:50.960651510 +0300
+++ nDPI/src/lib/protocols/ssl.c	2017-01-25 16:03:20.632967015 +0300
@@ -276,6 +276,10 @@
 			u_int begin = 0,len;
 			char *server_name = (char*)&packet->payload[offset+extension_offset];
 
+			if(packet->payload[offset+extension_offset+2] == 0x00) // host_name
+			{
+				begin =+ 5;
+			}
 			while(begin < extension_len) {
 			  if((!ndpi_isprint(server_name[begin]))
 			     || ndpi_ispunct(server_name[begin])
