diff -ruN PcapPlusPlus-master/Pcap++/src/DpdkDevice.cpp PcapPlusPlus-master1/Pcap++/src/DpdkDevice.cpp
--- PcapPlusPlus-master/Pcap++/src/DpdkDevice.cpp	2016-09-22 22:03:14.000000000 +0300
+++ PcapPlusPlus-master1/Pcap++/src/DpdkDevice.cpp	2016-09-29 21:40:35.000000000 +0300
@@ -19,6 +19,9 @@
 #define MBUF_SIZE (MBUF_DATA_SIZE + sizeof(struct rte_mbuf) + RTE_PKTMBUF_HEADROOM)
 
 #define RX_BURST_SIZE 64
+
+#define MEMPOOL_CACHE_SIZE 256
+
 
 namespace pcpp
 {
@@ -517,17 +520,12 @@
     bool ret = false;
 
     // Create transmission memory pool
-    memPool = rte_mempool_create(mempoolName, // The name of the mempool
-    		mBufPoolSize, // The number of elements in the mempool
-            MBUF_SIZE, // The size of each element
-            32, // cache_size
-            sizeof(struct rte_pktmbuf_pool_private),// The size of the private data appended after the mempool structure
-            rte_pktmbuf_pool_init, // A function pointer that is called for initialization of the pool
-            NULL, // An opaque pointer to data that can be used in the mempool
-            rte_pktmbuf_init, // A function pointer that is called for each object at initialization of the pool
-            NULL, // An opaque pointer to data that can be used as an argument
-            0, // socket identifier in the case of NUMA
-            MEMPOOL_F_SC_GET); // Flags
+	memPool = rte_pktmbuf_pool_create(mempoolName, // The name of the mempool
+		mBufPoolSize, // The number of elements in the mempool
+		MEMPOOL_CACHE_SIZE, // cache size
+		0, // 
+		RTE_MBUF_DEFAULT_BUF_SIZE, rte_socket_id());
+
 
     if (memPool == NULL)
 	{
